File: packages/point/python_std/point_std/json.py
from __future__ import annotations

import json
from typing import Any


def _compact_json(value: Any) -> str:
	return json.dumps(value, separators=(",", ":"), ensure_ascii=False)


def jsonParse(value: str) -> str | dict[str, str]:
	try:
		return _compact_json(json.loads(value))
	except Exception as error:
		return {"message": str(error)}


def jsonStringify(value: str) -> str:
	return _compact_json(json.loads(value))

File: packages/point/src/std/json.ts
export type PointStdError = { message: string };

export function jsonParse(value: string): string | PointStdError {
	try {
		return JSON.stringify(JSON.parse(value));
	} catch (error) {
		return { message: error instanceof Error ? error.message : String(error) };
	}
}

export function jsonStringify(value: string): string {
	return JSON.stringify(JSON.parse(value));
}
